optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -301,14 +305,16 @@ public void removeEventListener(SolverEventListener<Solution_> eventListener) {
     }
 
     public void addPhaseLifecycleListener(PhaseLifecycleListener<Solution_> phaseLifecycleListener) {
+        phaseLifecycleSupport.addEventListener(phaseLifecycleListener);
```

Comment:
There was and still is a design issue here:
This solver's phaseLifecycleSupport might be reused internally - as well as the phase's phaseLifecycleSupports. However currently the design presumes that any listener is in both (and that would mean that one created intended specifically for one phase will probably also get stepEnded events from other phases).

Because - by luck - neither the solver nor the phase phaseLifecycleSupport are reused beyond this method (which is used heavily by the statistics) - there are no bugs yet.

I'll refactor to separate them out and to have each phase also notify the "global" solver phaseLifecycleSupports upon phase/step events (but not on solver events to avoid duplication) by registering a bridge.